pennylane/devices/_legacy_device.py

```diff
@@ -784,12 +784,28 @@ def null_postprocess(results):
             # Hamiltonian/Sum with grouping indices already calculated.
             circuits, processing_fn = qml.transforms.split_non_commuting(circuit, "qwc")
 
-        elif any(isinstance(m.obs, (Hamiltonian, LinearCombination)) for m in circuit.measurements):
+        elif any(
+            isinstance(m.obs, (Hamiltonian, LinearCombination, Sum)) for m in circuit.measurements
+        ):
 
             # Otherwise, use wire-based grouping if the circuit contains a Hamiltonian
             # that is potentially very large.
             circuits, processing_fn = qml.transforms.split_non_commuting(circuit, "wires")
 
+            if all_obs_usable and not circuit.shots and len(circuits) > len(circuit.measurements):
```

Comment:
Doesn't this simply forces all legacy devices to support non-commuting measurements?